ash/src/extensions/ext/extended_dynamic_state.rs

```diff
@@ -0,0 +1,179 @@
+        let extended_dynamic_state_fn = vk::ExtExtendedDynamicStateFn::load(|name| unsafe {
+            mem::transmute(instance.get_device_proc_addr(device.handle(), name.as_ptr()))
+        });
+        ExtendedDynamicState {
+            handle: instance.handle(),
+            extended_dynamic_state_fn,
+        }
+    }
+
+    pub fn name() -> &'static CStr {
+        vk::ExtExtendedDynamicStateFn::name()
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetCullModeEXT.html>"]
+    pub unsafe fn cmd_set_cull_mode_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        cull_mode: vk::CullModeFlags,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_cull_mode_ext(command_buffer, cull_mode);
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetFrontFaceEXT.html>"]
+    pub unsafe fn cmd_set_front_face_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        front_face: vk::FrontFace,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_front_face_ext(command_buffer, front_face)
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetPrimitiveTopologyEXT.html>"]
+    pub unsafe fn cmd_set_primitive_topology_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        primitive_topology: vk::PrimitiveTopology,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_primitive_topology_ext(command_buffer, primitive_topology)
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetViewportWithCountEXT.html>"]
+    pub unsafe fn cmd_set_viewport_with_count_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        viewport_count: u32,
+        p_viewports: *const vk::Viewport,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_viewport_with_count_ext(command_buffer, viewport_count, p_viewports)
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetScissorWithCountEXT.html>"]
+    pub unsafe fn cmd_set_scissor_with_count_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        scissor_count: u32,
+        p_scissors: *const vk::Rect2D,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_scissor_with_count_ext(command_buffer, scissor_count, p_scissors)
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdBindVertexBuffers2EXT.html>"]
+    pub unsafe fn cmd_bind_vertex_buffers2_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        first_binding: u32,
+        binding_count: u32,
+        p_buffers: *const vk::Buffer,
+        p_offsets: *const vk::DeviceSize,
+        p_sizes: *const vk::DeviceSize,
+        p_strides: *const vk::DeviceSize,
```

Comment:
These should also take references or slices, where possible with proper validation that all slices share the same length. Note that `p_sizes` and `p_strides` can be optional.